src/passport.js

```diff
@@ -0,0 +1,68 @@
+const passwordGenerator = require('generate-password');
+const bcrypt = require('bcrypt');
+const { sendEmail } = require('./services/mail');
+const User = require('./models/user');
+
+const saltRounds = 10;
+
+const googleConfigs = {
+  clientID: process.env.GAPP_CLIENT_ID,
+  clientSecret: process.env.GAPP_CLIENT_SECRET,
+  callbackURL: 'http://localhost:3000/api/auth/google/callback',
+};
+
+const afterGoogleLogin = async function (
+  accessToken,
+  refreshToken,
+  profile,
+  cb
+) {
+  try {
+    let user = await User.findOne({ email: profile._json.email });
+    if (!user) {
+      const username = `${
```

Comment:
nice work generating a username, but I would suggest using this [package ](https://www.npmjs.com/package/unique-username-generator)which does the same work without needing to generate it manually

Reply:
Thank you for the suggestion. I just want to ask one question before implementing it, isn't it better to decrease the number of our external dependencies as much as possible? I mean as you said that the method I used will do the same job, since we need this dependency (password generator) why to add another dependency

Reply:
I agree with you, but I think such a small dependency won't make any difference here apart from increasing the readability of the code.

Reply:
Ok that's a good point, I will change it now. Thank you 😁